projects/packages/videopress/src/client/state/utils/map-videos.ts

```diff
@@ -12,3 +19,52 @@ export const mapVideo = ( video: OriginalVideoPressVideo ): VideoPressVideo => {
 export const mapVideos = ( videos: OriginalVideoPressVideo[] ): VideoPressVideo[] => {
 	return videos.map( mapVideo );
 };
+
+export const mapVideoFromWPV2MediaEndpoint = (
+	video: OriginalVideoPressVideo
+): VideoPressVideo => {
+	const { media_details: mediaDetails, id, caption } = video;
+
+	const { videopress: videoPressMediaDetails, width, height } = mediaDetails;
+
+	const {
+		title,
+		description,
+		original: url,
+		poster,
+		upload_date: date,
+		duration,
+		is_private: isPrivate,
+		file_url_base: fileURLBase,
+		files,
+	} = videoPressMediaDetails;
+
+	const { dvd } = files;
+
+	// Pick poster image from dvd file type.
+	const dvdImage = `${ fileURLBase.https }${ dvd.original_img }`;
```

Comment:
Just a point here, we could not rely on this info as the `posterImage`, since this info doesn't update when we change the poster, mainly because this as the name says, is the original poster image.

So, maybe is a case to change this comment and move this to a field more related to this data: `originalPosterImage` maybe?

Just an example of my current videos:

DVD Original Image: https://videos.files.wordpress.com/asKzPGrc/sample-video-2_mp4_dvd.original.jpg
Poster Image: https://videos.files.wordpress.com/asKzPGrc/sample-video-2_mp4.scrubthumb.jpg

Reply:
:TIL: let me take a look.

Reply:
I think the API should provide the updated poster for all video types. Something to improve form the VP client I guess.

Reply:
https://github.com/Automattic/jetpack/issues/26319